### PR TITLE
Adds connection listeners that can be called statically

### PR DESCRIPTION
**Pull request by ThadHouse**
This way you can have a connection listener before initializing NetworkTables

### CODE REVIEW

**Comment by PeterJohnson** [comment]
Would C++ benefit from this as well (for API similarity)?  Is this a breaking change?

**Comment by ThadHouse** [comment]
C++ does not have connection listeners at all in the NetworkTables API. And this should not be a breaking change. All of the IRemote functions are valid, and its always been up to the user to know exactly which instance the functions are being called on if the IRemote is received from the callbacks.

File: java/src/edu/wpi/first/wpilibj/networktables/NetworkTable.java [hunk 1]
```diff
@@ -243,6 +243,41 @@ public void apply(int uid, boolean connected, ConnectionInfo conn) {
...
+      return conns.length > 0;
+    }
+    public boolean isServer() {
+      return !client;
+    }
+  };
+  
+  private static final Hashtable<IRemoteConnectionListener,ConnectionListenerAdapter> globalConnectionListenerMap = new Hashtable<IRemoteConnectionListener,ConnectionListenerAdapter>();
+  public static synchronized void addGlobalConnectionListener(IRemoteConnectionListener listener,
+                                                       boolean immediateNotify) {
```
Comment (PeterJohnson) [inline java/src/edu/wpi/first/wpilibj/networktables/NetworkTable.java]: Indentation?

Comment (ThadHouse) [reply java/src/edu/wpi/first/wpilibj/networktables/NetworkTable.java]: Fixed

**Pull request merged**